### DISCUSSION TITLE
Right click -> Hold click

### DISCUSSION THREAD

**Discussion by Leblayd** [home-assistant/frontend#22149] [Accessibility]
I feel like a crazy person to have to ask for this, but I want Right Clicks to be usable.

Since the dawn of computers, left click has been the default interaction and right click the more advanced one.
This, when translated to mobile, became `tap` and `hold`.
HomeAssistant (or lovelace?) only has a mobile's functionality, even on PC.

I can't imagine 99% of the user base would never _ever_ use HASS on a PC. And I'm still not sure if this feature missing is perhaps a weird configuration issue.
(Although I did check the [demo](https://demo.home-assistant.io/#/lovelace/home) and it also does not have this feature)

I've _never_ used any other application that simply refuses to use the right click _but_ has basic interaction tied to holding a click.
I don't think I'm exaggerating when I say that it is the number 1 reason I think about abandoning my entire smart home for.

Surely I'm doing something wrong because I don't see a single complaint about it on here, nor on the forums or even on reddit. So, surely, it must actually work, right?

**Comment by ildar170975** [comment]
Right click on a browser’s page is supposed to show a browser’s context menu.

**Comment by Leblayd** [comment]
Sure, as it's a built-in function of the browser. Which, to be fair, is the standard for just about all web apps.

But the problem is: most web-apps therefore completely _avoid_ using any sort of "secondary click".
HASS does not: it has long presses on mobile. So I think it should have a platform equivalent action on desktop.
Because "long clicking" on desktop feels absolutely terrible. It's such a weird concept that there isn't even a word for it AFAIK.

But this is _only_ because most web-apps aren't complex enough to require it, and find a different way to solve it (usually an ellipsis menu). HASS very much _is_ complex enough. And the ellipsis menu solution is already taken.
(Although I would also support a functional context menu just as much. But that's an entirely separate issue.)

Think of something like Google Drive, which has a similarly feature-rich UI.
On desktop: right click brings up a context menu. Holding a left click is not an option.
On mobile: long pressing selects the item, so that you can perform the _same_ context dependent actions on it
 
 In this case, though, because lovelace has configurable options for Regular and Hold clicks, I think the clear best solution (or at least first step) would be to have an option so that the right click emulates that.

Edit:
Looked at what HASS actually does. It seems it's already [listening to Right Clicks](https://github.com/home-assistant/frontend/blob/4c2044e70a257aa0c90d53d8ad99c6928b33affc/src/panels/lovelace/common/directives/action-handler-directive.ts#L113), and it's actually even preventing the context menu from popping up on elements that can be held.

From what I can see (this is literally my first time looking into the code base), then this should be about it:
```ts
if (options.hasHold) {
    fireEvent(target, "action", { action: "hold" });
}
```

I was hoping to write a simple js script that can fire the event for me, but as the elements prevent the bubbling, I don't even have a chance to catch it on the document.

**Comment by Merlin04** [comment]
thank you so much for posting this, i'm glad i'm not the only one frustrated by this!! i just installed home assistant for the first time and couldn't figure out how to hide a dashboard from the sidebar, and eventually found a forum thread saying i needed to click and hold on the "home assistant" text in the sidebar to get a dialog box where i could do that. there is zero chance i ever could have figured that out on my own, there's no indication that the header text is an interactable element. i also noticed that the web ui disables the browser right click menu on "long-click" elements so they could have easily made right click map to "long click", there's really no reason to not do this.

**Reply by karwosts** [reply]
> ! i just installed home assistant for the first time and couldn't figure out how to hide a dashboard from the sidebar, and eventually found a forum thread saying i needed to click and hold on the "home assistant" text in the sidebar to get a dialog box where i could do that. there is zero chance i ever could have figured that out on my own, there's no indication that the header text is an interactable element

While yes, that is an odd little easter-egg shortcut, to be fair it's not the only way to do this. The same menu is available from the more appropriately accessible menu in the user preferences, which requires no goofy hold actions😅 

<img width="586" height="75" alt="image" src="https://github.com/user-attachments/assets/0e023e0b-0bd2-4328-88d3-510e44200e35" />

**Reply by Leblayd** [reply]
> there is zero chance i ever could have figured that out on my own, there's no indication that the header text is an interactable element.

Can confirm, because I didn't even know about that 'easter egg's and I've been using Hass for a fairly long time now. (I did know the alternative way to configure it, of course.)

It goes to show how incredibly unintuitive this long click-based design is.
